src/main/java/com/amazon/opendistroforelasticsearch/sql/query/ESActionFactory.java

```diff
@@ -60,9 +60,9 @@
      * @return Query object.
      */
     public static QueryAction create(Client client, String sql) throws SqlParseException, SQLFeatureNotSupportedException {
-        sql = sql.replaceAll("\n"," ");
-        String firstWord = sql.substring(0, sql.indexOf(' '));
-        switch (firstWord.toUpperCase()) {
+        sql = sql.replaceAll("\n"," ").trim();
```

Comment:
I think if `sql.isEmpty()` after this statement, there's no need to go into that switch, and we need to throw a different exception, most likely something like invalid query/request (I am sure we already have something like that). `SQLFeatureNotSupportedException` indicates a case where you might have a valid sql statement, but we can't handle it. I think it's better to put a clear distinction between those two cases, at least start with some case.

Reply:
another note, should we replace `String.format(Locale.ROOT, "%n")` instead of `"\n"`? What if the user sends `"\n\r"`?

Reply:
1st comment: I agree with the comment that chosen exception is misleading, I wanted to limit the scope of change. (SqlParseException might be a better choice here).

2nd comment: agree, will change to System.lineSeparator()

Reply:
> 1st comment: I agree with the comment that chosen exception is misleading, I wanted to limit the scope of change. (SqlParseException might be a better choice here).
> 
> 2nd comment: agree, will change to System.lineSeparator()

For 2nd comment, just curious, where/when this separator added to the original query? I assume changing it to `System.lineSeparator()` means use separator of platform our plugin is running on right? I wonder is it necessarily same as the separator in query we want to get rid of?

Reply:
@dai-chen , that's a good point, those are not necessarily the same, so I think we should just replace all types of new lines.

Reply:
> @dai-chen , that's a good point, those are not necessarily the same, so I think we should just replace all types of new lines.

Yeah, that was my concern.

Reply:
@galkk does this make sense to you?

Reply:
Done